docs/coding-conventions.md

```diff
@@ -0,0 +1,297 @@
+      in packages/files with representative names.
+- Prefer to define variables in the function body
+    - Avoid returning too many values from a function.
+    - Avoid naked returns.
+- Use closures with caution, be aware of the risks and use them only when it
+  makes sense.
+    - They can still be used if there is a good reason for it.
+    - A useful use case of closures is to define a function that is relevant
+      only for a very specific scope.
+- Declare empty slices with the var syntax.
+    - Pay attention when serializing data, see [Declaring Empty Slices](https://go.dev/wiki/CodeReviewComments#declaring-empty-slices).
+- Avoid use of `fmt.Sprintf` for manual construction of complex objects
+  or operations (e.g. paths or patches).
+    - Use helpers or builders when available.
+        - Build patches with the `PatchSet` interface.
+        - Construct paths with the `path` package.
+- Use the `kubevirt.io/kubevirt/pkg/pointer` package when pointers are
+  required.
+- Keep function signatures lean.
+    - E.g. use `kubevirt.Client()` in test functions instead of passing the
+      client as an additional function call argument.
+    - Again: Use structs and receiver methods to keep state.
+- Table-driven tests are preferred for testing matrices of scenarios/inputs.
+    - Use Gingko's `DescribeTable` to construct test tables.
+- Do not expect an asynchronous thing to happen immediately.
+    - For example do not wait for one second and expect a VM to be running.
+    - Wait and retry instead, in test code use `Eventually`.
+- Avoid `Skip` in tests.
+    - Use `decorators` to control in which lanes tests run.
+- Namings (packages, interfaces, etc.)
+    - Consider the package name when selecting an interface name and avoid
+      redundancy.
+      - For example, `storage.Interface` is better than `storage.
+        StorageInterface`.
+    - Consider the parent directory name when choosing a package name.
+      - For example, `pkg/controllers/autoscaler/foo.go` should
+        say `package autoscaler` not `package autoscalercontroller`.
+    - Unless there's a good reason, the `package foo` line should match the
+      name of the directory in which the `.go` file exists.
+    - Do not use uppercase characters, underscores, or dashes in package names.
+    - Command-line flags should use dashes, not underscores.
+    - Importers can use a different name if they need to disambiguate.
+    - Locks should be called `lock` and should never be embedded (
+      always `lock sync.Mutex`).
+    - When multiple locks are present, give each lock a distinct name following
+      Go conventions: `stateLock`, `mapLock` etc.
+- Avoid relying on Docker Hub.
+    - Use the [Google Cloud Container Registry](https://gcr.io) instead.
+
+## Directory and file conventions
+
+- Avoid package sprawl. Find an appropriate subdirectory for new packages.
+    - Libraries with no appropriate home belong in new package subdirectories
+      of `pkg/util`.
```

Comment:
Ohh no, they do not. This is another version of `utils.go`.
They belong to a new properly named package/library.

If no "home" can be found, consult with your reviewers.